addons/multi_company_guidelines/views/views.xml

```diff
@@ -0,0 +1,156 @@
+<odoo>
+  <data>
+    <!-- shared model -->
+    <record model="ir.ui.view" id="shared_list">
+      <field name="name">multi_company_guidelines shared list</field>
+      <field name="model">multi_company.shared</field>
+      <field name="arch" type="xml">
+        <tree string="Shared Model" editable="bottom">
+          <field name="name" />
+          <field name="company_id" />
+          <field name="property_field" />
+        </tree>
+      </field>
+    </record>
+
+    <!-- main model -->
+    <record model="ir.ui.view" id="main_list">
+      <field name="name">multi_company_guidelines main list</field>
+      <field name="model">multi_company.main</field>
+      <field name="arch" type="xml">
+        <tree string="Main Model">
+          <field name="name" />
+          <field name="company_id" />
+        </tree>
+      </field>
+    </record>
+
+    <record id="main_form" model="ir.ui.view">
+      <field name="name">multi_company_guidelines main form</field>
+      <field name="model">multi_company.main</field>
+      <field name="arch" type="xml">
+        <form string="Main Model">
+          <header>
+            <button type="object" name="action_draft" string="Reset to Draft" states="confirmed" />
+            <button type="object" name="action_confirm" string="Confirm" states="draft" class="oe_highlight" />
+            <button type="object" name="action_post" states="confirmed" string="Post" class="oe_highlight" />
+            <field name="state" widget="statusbar" />
+          </header>
+          <sheet>
+            <!--<div class="oe_button_box" name="button_box">
+              <button name="action_view_invoices" type="object" class="oe_stat_button" icon="fa-money"
+                attrs="{'invisible': [('invoice_ids_nbr', '=', 0)]}">
+                <field name="invoice_ids_nbr" widget="statinfo" string="Invoice(s)" />
+              </button>
+            </div>-->
+            <group>
+              <group>
+                <field name="partner_id"/>
```

Comment:
this many2one is not filtered in the model nor the views
-> i'm in company A
-> i create main record, its company_d is A
-> i tick the access to company ABC
-> i chose a partner from company C
-> i confirm this main record
-> consistency error

IMO every m2o should be filtered according to the company of the main record